lms/static/js/student_account/views/AccessView.js

```diff
@@ -264,6 +264,12 @@
 
                 // Load the form. Institution login is always refreshed since it changes based on the previous form.
                     if (!this.form.isLoaded($form) || type == 'institution_login') {
+
+                        // We need a special case for loading reset form as there is mismatch of form id
+                        // value ie 'password-reset' vs load function name ie 'reset'
+                        if (type === 'password-reset') {
```

Comment:
As discussed this is necessary as how template for reset page is using 'password-reset' and not 'reset' as form type. Previous code was implemented in a way to directly call loadForm function passing `reset` after handling mouse click on button and that is different from registration and login page. This is done to avoid too much changing to template structure that might cause unseen ripple effects and possibly acceptance test to fail.